src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -607,5 +608,6 @@ public enum ActivityIdFormat : byte
         Unknown,      // ID format is not known.     
         Hierarchical, //|XXXX.XX.X_X ... see https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/ActivityUserGuide.md#id-format
         W3C,          // 00-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX-XXXXXXXXXXXXXXXX-XX see https://w3c.github.io/trace-context/
+        ForceW3C,     // This value is only used by DefaultIdFormat.   It indicate the W3C format bit it also indicate W3C should be preferred over the parent ID format.  
```

Comment:
> Scenario level - Assume I own service A which calls out to service B. I want to use W3C in A, but I can't control B and it uses Hierarchical. Is the expectation that I should be able to have a trace that covers both with some type of interop in the middle? If yes, the most obvious way of doing that interop is creating an Activity with a hierarchial ID format before calling into the http client code, but I won't be able to do that if ForceW3C is set.

To achieve back-compat we have to emit both: W3C and hierarchical in corresponding headers. This will satisfy all possible scenarios. We've discussed it a bit and my proposal is to inject Request-Id with hierarchical Id on ApplicaitonInsights level - we'll gradually deprecate it.